### PR TITLE
fix persistence agent to use in cluster DNS instead of kube proxy to access API

### PR DESCRIPTION
**Pull request by IronPan**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/538)

### CODE REVIEW

File: backend/src/agent/persistence/client/pipeline_client.go [hunk 1]
```diff
@@ -44,38 +44,32 @@ type PipelineClient struct {
...
-	err := util.WaitForGrpcClientAvailable(namespace, initializeTimeout, basePath, masterurl,
-		kubeconfig)
+		namespace string,
+		initializeTimeout time.Duration,
+		timeout time.Duration,
+		basePath string,
+		mlPipelineServiceName string,
+		mlPipelineServiceHttpPort string,
+		mlPipelineServiceGRPCPort string) (*PipelineClient, error) {
+	addressTemp := "%s.%s.svc.cluster.local:%s"
```
Comment (vicaire) [inline backend/src/agent/persistence/client/pipeline_client.go]: Could you make this a constant?

**Comment by vicaire** [comment]
/lgtm

One small comment. Feel free to approve yourself.

File: backend/src/agent/persistence/client/pipeline_client.go (hunk 1 above)
Comment (IronPan) [reply backend/src/agent/persistence/client/pipeline_client.go]: done

**Pull request merged**